jpsonic-main/src/test/java/com/tesshu/jpsonic/dao/MediaFileDaoTest.java

```diff
@@ -26,613 +26,702 @@
-                    null, null, false, false, null);
-            builder = new RandomSongsQueryBuilder(criteria);
-            query = builder.build();
-            assertTrue(assertWithCondition.apply(query));
-
-            // maxLastPlayedDateCondition
-            criteria = new RandomSearchCriteria(0, null, null, null, Collections.emptyList(), null, now(), null, null,
-                    null, null, false, false, null);
-            builder = new RandomSongsQueryBuilder(criteria);
-            query = builder.build();
-            assertTrue(assertWithCondition.apply(query));
-
-            // maxAlbumRatingCondition
-            criteria = new RandomSearchCriteria(0, null, null, null, Collections.emptyList(), null, null, 1, 2, null,
-                    null, false, false, null);
-            builder = new RandomSongsQueryBuilder(criteria);
-            query = builder.build();
-            String suffix = SELECT + CONDITION_JOIN2 + WHERE;
-            assertTrue(query.startsWith(suffix));
-            assertTrue(query.endsWith(ORDER_LIMIT));
-            assertTrue(suffix.length() < query.length());
-
-            // minPlayCountCondition
-            criteria = new RandomSearchCriteria(0, null, null, null, Collections.emptyList(), null, null, null, null, 1,
-                    null, false, false, null);
-            builder = new RandomSongsQueryBuilder(criteria);
-            query = builder.build();
-            assertTrue(assertWithCondition.apply(query));
-
-            // maxPlayCountCondition
-            criteria = new RandomSearchCriteria(0, null, null, null, Collections.emptyList(), null, null, null, null,
-                    null, 1, false, false, null);
-            builder = new RandomSongsQueryBuilder(criteria);
-            query = builder.build();
-            assertTrue(assertWithCondition.apply(query));
-
-            // showUnstarredSongsCondition
-            criteria = new RandomSearchCriteria(0, null, null, null, Collections.emptyList(), null, null, null, null,
-                    null, null, false, true, null);
-            builder = new RandomSongsQueryBuilder(criteria);
-            query = builder.build();
-            suffix = SELECT + CONDITION_JOIN1 + WHERE;
-            assertTrue(query.startsWith(suffix));
-            assertTrue(query.endsWith(ORDER_LIMIT));
-            assertTrue(suffix.length() < query.length());
+        void testGetMudicIndexCounts() {
+            List<MusicFolder> folders = List.of(MUSIC_FOLDER);
+            List<IndexWithCount> counts = mediaFileDao.getMudicIndexCounts(folders, Collections.emptyList());
+            assertEquals(2, counts.size());
+            counts.stream().forEach(index -> {
+                switch (index.index()) {
+                case "D" -> assertEquals(2, index.artistCount());
+                case "" -> assertEquals(1, index.artistCount());
```

Comment:
This case is not good.  It has not been decided how to index the Album directory that exists directly under the music folder. (That's a very irregular case though.)

Reply:
Therefore, the following modifications are necessary.

 - In the above test case, modify so that Album directly under MusicFolder is indexed.
 - Fix the field name of Record used in MediaFileDao

Reply:
Done.